Readme.md

```diff
@@ -241,15 +241,18 @@ metrics:
   # A map of string to string for constant labels. This labels will be attached to every prometheus metric
    const_labels:
     sensor_type: ikea
+  # When specified, metric value to use if a value cannot be parsed (match cannot be found in the map above, invalid float parsing, expression fails, ...)
+  # If not specified, parsing error will occur.
+  error_value: 1
+  # When specified, if the value cannot be parsed or expression fails, runs this expression as fallback
+  # if it fails, error_value will be returned
+  error_expression: "float(join(filter(split(a, ''), { # matches '^[0-9]$' }), ''))"
```

Comment:
I wonder why we need an error_expression? Can't we just use a normal expression to implement parsing of strings like `H20241227104456`?  The only reason I understand are message producers writing messages in inconsistent formats so that we have to apply two different parsing strategies. Is that the case?

Reply:
using `expression` is not possible because it runs after type conversion and it fails as it's not a `bool`, a `float` nor a static map string

I can see several options:
- make expression to run before type conversion. but this would break all current expression implementation because the `value` sent is the value after type conversion
- create another expression that runs before type conversion. Like `expression_raw` and if set, return a float from whatever value is read from MQTT bypassing the current type conversion/expression
- add error_expression which is kind of the same of the previous one but seems more generic.

Reply:
I actually would like to prever the second solution:
> create another expression that runs before type conversion. Like expression_raw and if set, return a float from whatever value is read from MQTT bypassing the current type conversion/expression

This is more logical having a parsing expression yourself instead of doing it after the fact. In that case, if the expression returns a string, we could enable the user to parse the string first, and then apply the mapping the user specified.

Reply:
agree,

i'll push another PR with raw_expression and only keep error_value in this one

Reply:
see #161 and #162